### PR TITLE
#167733825 Integrate The Request Table With The Backend

### PR DESCRIPTION
**Pull request by EziamakaNV**
#### What does this PR do?
- This pull request creates the models and migrations of the requests and trips with their corresponding associations/references

#### Description of Task to be completed?
- Create the models and migrations using sequelize
- Configure the corresponding associations/references
- Implement an endpoint to query for all requests pertaining to a particular user
- Write a unit test for the endpoint

#### How should this be manually tested?
- Ensure you have git, node and  npm installed
- Clone the colossus-backend repo
- Open your terminal and navigate to the repo you've just cloned
- Enter the command 'npm install && npm test'

#### Any background context you want to provide?
- A user can make multiple requests and a request can have multiple trips tied to it

#### What are the relevant pivotal tracker stories?
- #167733825

### CODE REVIEW

**Comment by tamunoibi** [comment]
Great job.

However, I noticed in
src/database/migrations/20190827211052-create-request.js line 27. 

You have managersId. That field should not be on the request table. It should be on the user's table.  This is managersId as a property of users
`users {
  first_name: 'John Micheal,
  managerId: 5769' ,
}`

This is managersId as a property of request
`requests {
  type: trip,
  from: 'Nigeria',
  to: 'Korea',
  managersId: 456,
}`

**Comment by EziamakaNV** [comment]
> Great job.
> 
> However, I noticed in
> src/database/migrations/20190827211052-create-request.js line 27.
> 
> You have managersId. That field should not be on the request table. It should be on the user's table. This is managersId as a property of users
> `users { first_name: 'John Micheal, managerId: 5769' , }`
> 
> This is managersId as a property of request
> `requests { type: trip, from: 'Nigeria', to: 'Korea', managersId: 456, }`

Hi Ib, remember during the last meeting where the team went through the PT board we all agreed that the user would not know their manager on signup, that they would have to input the manager detail when requesting a trip.

I know you're working on the manager notifications feature. Does your task require the managerId column to be in the user model?

**Review by fegzycole — APPROVED** [review]
Looks good

**Comment by kingsley-einstein** [comment]
Good job. But could you explain why you removed some configuration details?
Also, why do you need "fs" module for sequelize?

**Comment by EziamakaNV** [comment]
> Good job. But could you explain why you removed some configuration details?
> Also, why do you need "fs" module for sequelize?

I edited the file to how it was originally before I pulled your PR recently from develop. I was experiencing issues with my models. The entire file is created by sequelize. What the "fs" part of the code does is to read all the files in the models folder to configure the models and setup the associations between models and then export them

File: src/test/test.js [hunk 1]
```diff
@@ -25,14 +25,14 @@ let verificationToken;
...
-      force: true
-    })
-      .then(() => {
-        done();
-      });
-  });
+  // before((done) => {
+  //   sequelize.sync({
+  //     force: true
+  //   })
```
Comment (Oluwafayokemi) [inline src/test/test.js]: Why is there a commented code here?

Comment (EziamakaNV) [reply src/test/test.js]: Hi Fayo, I've just made the changes

**Review by Oluwafayokemi — APPROVED** [review]

**Pull request merged**